tools/osv-scanner/internal/output/machinejson.go

```diff
@@ -0,0 +1,71 @@
+package output
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+
+	"github.com/google/osv.dev/tools/osv-scanner/internal/osv"
+)
+
+type Output struct {
+	Results []Result `json:"results"`
+}
+
+type Result struct {
+	FilePath string    `json:"filePath"`
+	Packages []Package `json:"packages"`
+}
+
+type Package struct {
+	Name            string              `json:"name"`
+	Version         string              `json:"version"`
+	Ecosystem       string              `json:"ecosystem"`
+	Vulnerabilities []osv.Vulnerability `json:"vulnerabilities"`
+}
+
+// PrintJSONResults writes results to the provided writer in JSON format
+func PrintJSONResults(query osv.BatchedQuery, resp *osv.HydratedBatchedResponse, outputWriter io.Writer) {
+	output := Output{}
+	groupedBySource := map[string][]Package{}
+
+	for i, query := range query.Queries {
+		response := resp.Results[i]
+		if len(response.Vulns) == 0 {
```

Comment:
I wonder if we need a flag where we output everything we scanned, just to provide assurance that we did in fact consider all the dependencies that were provided. Can come later if people ask for it.